jdk/src/share/classes/java/util/ResourceBundle.java

```diff
@@ -1398,7 +1398,13 @@ private static ResourceBundle getBundleImpl(String baseName, Locale locale,
             bundle = baseBundle;
         }
 
-        return bundle;
+        //The OpenJ9 GC might collect the loader before we return here. This prevents that.
+        try {
+            return bundle;
+        } finally {
+            //Should never be true. Using the loader here prevents it being GC'd.
+            if (baseName.equals(loader)) throw new Error("Unexpected error.");
```

Comment:
This could use == instead of .equals(), which should be a faster test.

Reply:
Incomparable object. Will use a global Object for the compare.